packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -297,6 +297,7 @@ exports.createContentTypeNodes = ({
           const entryItemFieldValue = entryItemFields[entryItemFieldKey]
           if (Array.isArray(entryItemFieldValue)) {
             if (
+              entryItemFieldValue[0] &&
```

Comment:
What do you think about moving this check a bit up?

`if (Array.isArray(entryItemFieldValue) && entryItemFieldValue.length > 0) {`

Reply:
This sounds great! you can even make it shorter by omitting the `> 0`

`if (Array.isArray(entryItemFieldValue) && entryItemFieldValue.length) {`

Reply:
Yeah, I did that initially. But if `entryItemFieldValue` is indeed an array, does it make sense to jump to the `else if` below?

I was going for the smallest change possible, otherwise maybe we could make this block a little DRYier:

```js
const value = Array.isArray(entryItemFieldValue)
  ? entryItemFieldValue
  : [entryItemFieldValue]

value
  .filter(v => _.has(v, 'sys.id'))
  // Don't create link to an unresolvable field.
  .filter(v => resolvable.has(v.sys.id)) 
  .forEach(v => {
    if (!foreignReferenceMap[v.sys.id]) {
      foreignReferenceMap[v.sys.id] = []
    }

    foreignReferenceMap[v.sys.id].push({
      name: `${contentTypeItemId}___NODE`,
      id: entryItem.sys.id,
    })
  })
```

Reply:
Actually, I am wrong @brosapinto, sorry about that I totally missed the else if